packages/protocol/migrations/17_elect_validators.ts

```diff
@@ -139,48 +139,81 @@ module.exports = async (_deployer: any) => {
   }
 
   if (valKeys.length < config.validators.minElectableValidators) {
-    console.info(
+    console.warn(
       `  Warning: Have ${valKeys.length} Validator keys but require a minimum of ${
         config.validators.minElectableValidators
       } Validators in order for a new validator set to be elected.`
     )
   }
 
-  console.info('  Registering ValidatorGroup ...')
-  const firstPrivateKey = valKeys[0]
-  const account = await registerValidatorGroup(lockedGold, validators, firstPrivateKey)
+  // Split the validator keys into groups that will fit within the max group size.
+  const valKeyGroups: string[][] = []
+  const maxGroupSize: number = Number(config.validators.maxGroupSize)
+  for (let i = 0; i < valKeys.length; i += maxGroupSize) {
+    valKeyGroups.push(valKeys.slice(i, Math.min(i + maxGroupSize, valKeys.length)))
+  }
 
-  console.info('  Registering Validators ...')
-  await Promise.all(
-    valKeys.map((key) => registerValidator(lockedGold, validators, key, account.address))
-  )
+  let prevGroupAddress = NULL_ADDRESS
+  for (const [idx, groupKeys] of valKeyGroups.entries()) {
+    // Append an index to the group name if there is more than one group.
+    let groupName: string = config.validators.groupName
+    if (valKeyGroups.length > 1) {
+      groupName += ` (${idx + 1})`
+    }
 
-  console.info('  Adding Validators to Validator Group ...')
-  for (let i = 0; i < valKeys.length; i++) {
-    const key = valKeys[i]
-    const address = generateAccountAddressFromPrivateKey(key.slice(2))
-    if (i === 0) {
-      // @ts-ignore
-      const addTx = validators.contract.methods.addFirstMember(address, NULL_ADDRESS, NULL_ADDRESS)
-      await sendTransactionWithPrivateKey(web3, addTx, account.privateKey, {
-        to: validators.address,
-      })
-    } else {
-      // @ts-ignore
-      const addTx = validators.contract.methods.addMember(address)
-      await sendTransactionWithPrivateKey(web3, addTx, account.privateKey, {
-        to: validators.address,
-      })
+    console.info(`  Registering Validator Group: ${groupName} ...`)
+    const account = await registerValidatorGroup(groupName, lockedGold, validators, groupKeys[0])
+
+    console.info('  * Registering Validators ...')
+    for (const key of groupKeys) {
+      await registerValidator(lockedGold, validators, key, account.address)
```

Comment:
Could you add a log here? This is where I was unsure if things had broken because it was hanging so long, only by looking at ethstats/blockscout was it clear that things were still happening

Reply:
I've added some log statements 👍